logger/logger.go

```diff
@@ -54,15 +54,30 @@ func SetLogger(zl *zap.Logger) {
 	logger = &Logger{zl.Sugar()}
 }
 
+var JSONLogFileName = "log.jsonl"
+
+// ProductionLoggerFilepath returns the full path to the file the
+// ProductionLogger logs to.
+func ProductionLoggerFilepath(configRootDir string) string {
+	return path.Join(configRootDir, JSONLogFileName)
+}
+
 // CreateProductionLogger returns a log config for the passed directory
 // with the given LogLevel and customizes stdout for pretty printing.
-func CreateProductionLogger(dir string, jsonStdout bool, lvl zapcore.Level) *zap.Logger {
+func CreateProductionLogger(
+	dir string, jsonStdout bool, lvl zapcore.Level, toDisk bool) *zap.Logger {
 	config := zap.NewProductionConfig()
-	destination := path.Join(dir, "log.jsonl")
 	if !jsonStdout {
-		config.OutputPaths = []string{"pretty", destination}
+		config.OutputPaths = []string{"pretty"}
+	}
+	config.ErrorOutputPaths = []string{"stderr"}
+	if toDisk {
+		destination := ProductionLoggerFilepath(dir)
+		if !jsonStdout {
```

Comment:
So I did some research into `jsonStdout` and thought about its coexistence with `logToDisk`. It was originally implemented as such in SHA ec00ca4c:

```diff
-       config.OutputPaths = []string{"pretty", destination}
+       if !jsonStdout {
+               config.OutputPaths = []string{"pretty", destination}
+       }
```

This is simply turning off all customizations and relying on the defaults:

```golang
		OutputPaths:      []string{"stderr"}, // relevant
		ErrorOutputPaths: []string{"stderr"}, // kept here just as an FYI
```

The goal of this story is to turn off writing logs to disk. The goal of `jsonStdout` is to only write json only to `stdout` (altho it's actually writing to `stderr`) therefore turning off logs to disk.

To recap:

* We always write json to the log file
* We pretty print to stderr if `jsonStdout` is false
* We write json to stderr if `jsonStdout` is true
* What should happen if they set `jsonStdout` and `logToDisk`? Right now, only errors are logged to disk. I think `jsonStdout` should only impact the formatting of the console output and not dictate whether or not we're logging to disk.

Refactor suggestion:

```golang
pretty := !jsonStdout
if pretty {
  config.OutputPaths = []string{"pretty"}
}
if toDisk {
  config.OutputPaths = append(config.OutputPaths, destination)
}
```

I'm also open to the idea of renaming `JSONStdout`, but let's talk about that at standup.

```diff
-	JSONStdout               bool            `env:"JSON_STDOUT" envDefault:"false"`
+	JSONConsole             bool            `env:"JSON_CONSOLE" envDefault:"false"`
```

Reply:
Thanks for the suggestion. This is 6880b7c:

```diff
 	config.ErrorOutputPaths = []string{"stderr"}
 	if toDisk {
 		destination := ProductionLoggerFilepath(dir)
-		if !jsonStdout {
-			config.OutputPaths = append(config.OutputPaths, destination)
-		}
+		config.OutputPaths = append(config.OutputPaths, destination)
 		config.ErrorOutputPaths = append(config.ErrorOutputPaths, destination)
 	}
 	config.Level.SetLevel(lvl)
```

I think we still want to log errors to disk.

Reply:
cff87d83